### PR TITLE
Implement CommentsHeader component

### PR DESCRIPTION
**Pull request by EVAST9919**

### CODE REVIEW

File: osu.Game/Overlays/Comments/SortSelector.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+                protected override void OnHoverLost(HoverLostEvent e)
+                {
+                    if (!Active.Value) base.OnHoverLost(e);
+                }
+            }
+        }
+    }
+
+    public enum SortCommentsBy
```
Comment (iiSaLMaN) [inline osu.Game/Overlays/Comments/SortSelector.cs]: this name seems inconsistent to me, what about `CommentSortCriteria` or just `SortCriteria`

**Review by frenzibyte — CHANGES_REQUESTED** [review]

File: osu.Game/osu.Game.csproj [hunk 2]
```diff
@@ -32,4 +32,7 @@
     <PackageReference Include="SharpRaven" Version="2.4.0" />
     <PackageReference Include="System.ComponentModel.Annotations" Version="4.6.0" />
   </ItemGroup>
+  <ItemGroup>
+    <Folder Include="Overlays\Comments\" />
+  </ItemGroup>
```
Comment (bdach) [inline osu.Game/osu.Game.csproj]: There doesn't seem to be a need for this to land in the `.csproj`.

**Review by bdach — COMMENTED** [review]
Visual reference:

image

Looks to be in line with web design and working properly - I just have one comment on the `.csproj` change which seems to be unnecessary.

File: osu.Game/osu.Game.csproj (hunk 2 above)
Comment (EVAST9919) [reply osu.Game/osu.Game.csproj]: It was an auto addition, will remove

**Review by bdach — DISMISSED** [review]
Looks good to go as far as I can tell 👍

**Comment by peppy** [comment]
Just as a note, there's basically zero point in making constants for values which only appear once. The exception is if it's not easy to know what the constant *is*, but in all the cases in your new class, they are used next to a keyword which defines them:

image

**Comment by peppy** [comment]
@EVAST9919 i've made several refactors to improve code quality, please make sure to review them.

**Review by peppy — APPROVED** [review]

**Pull request merged**